amundsen_application/static/js/components/common/ResourceList/PaginatedApiResourceList/index.spec.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import Pagination from 'react-js-pagination';
+
+import { shallow } from 'enzyme';
+
+import { ResourceType } from 'interfaces';
+import ResourceListItem from 'components/common/ResourceListItem/index';
+
+import PaginatedApiResourceList, { PaginatedApiResourceListProps } from './';
+import * as CONSTANTS from '../constants';
+
+describe('PaginatedApiResourceList', () => {
+  const setStateSpy = jest.spyOn(PaginatedApiResourceList.prototype, 'setState');
+  const setup = (propOverrides?: Partial<PaginatedApiResourceListProps>) => {
+    const props: PaginatedApiResourceListProps = {
+      activePage: 3,
+      itemsPerPage: 4,
+      onPagination: jest.fn(),
+      slicedItems: [
+        { type: ResourceType.table },
+        { type: ResourceType.table },
+        { type: ResourceType.table },
+        { type: ResourceType.table },
+      ],
+      totalItemsCount: 40,
+      source: 'testSource',
+      ...propOverrides
+    };
+    const wrapper = shallow<PaginatedApiResourceList>(<PaginatedApiResourceList {...props} />);
+    return { props, wrapper };
+  };
+
+  describe('onPagination', () => {
+    it('calls the onPagination prop', () => {
+      const setupResult = setup();
+      const wrapper = setupResult.wrapper;
+      const props = setupResult.props;
+      const onPaginationSpy = jest.spyOn(props, 'onPagination');
+      wrapper.instance().onPagination(3);
+      expect(onPaginationSpy).toHaveBeenCalledWith(2);
+    });
```

Comment:
It is usually recommended to call `wrapper.unmount();` after every jest test case to keep the scope clean and avoid weird and hard to debug bugs.

Some teams prefer to do it on every test case, some prefer to make the `wrapper` variable a shared one and do it on a `beforeEach` clause.

Reply:
This is new to me. I'd see the benefit if we were using `mount` for the tests,  but is it something that we have to do event with `shallow` rendering?

Reply:
Good call!
It can definitely be done, as [in the docs they use it as an example](https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/unmount.html), but a quick search didn't give me any insights.

Not a big deal though!